lighthouse-core/gather/computed/metrics/lantern-max-potential-fid.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const LanternMetricArtifact = require('./lantern-metric');
+const BaseNode = require('../../../lib/dependency-graph/base-node');
+
+/** @typedef {BaseNode.Node} Node */
+
+class LanternMaxPotentialFID extends LanternMetricArtifact {
+  get name() {
+    return 'LanternMaxPotentialFID';
+  }
+
+  /**
+   * @return {LH.Gatherer.Simulation.MetricCoefficients}
+   */
+  get COEFFICIENTS() {
+    return {
+      intercept: 0,
+      optimistic: 0.4,
+      pessimistic: 0.4,
```

Comment:
whats this mean? the two graphs are the same. Both are adjusted?

similarly, lantern-speed-index has two identical graphics but then has different coefficients. 

![](https://media0.giphy.com/media/1M9fmo1WAFVK0/giphy.gif)

Reply:
The graphs aren't the differentiating part of the optimistic/pessimistic equation here, it's actually whether we use optimistic/pessimistic FCP as the start point :)

The 0.4 was to mirror EIL, but honestly the justification for it (that we'll overestimate EIL a tad) hasn't proved to be true in the data, so I'll stick to 0.5 (simple average of optimistic + pessimistic estimates)


For speed index case the "optimistic"/"pessimistic" labels are a real stretch. The "optimistic" one is just normal speed index on the fast trace, the graph is essentially ignored completely here. The "pessimistic" one is our lantern psuedo-speed index based on the TTI graph.